controllers/predicate/predicate.go

```diff
@@ -190,12 +161,25 @@ func IsSnapshotLease() predicate.Predicate {
 		return strings.HasSuffix(lease.Name, "full-snap") || strings.HasSuffix(lease.Name, "delta-snap")
 	}
 
+	holderIdentityChange := func(objOld, objNew client.Object) bool {
+		leaseOld, ok := objOld.(*coordinationv1.Lease)
+		if !ok {
+			return false
+		}
+		leaseNew, ok := objNew.(*coordinationv1.Lease)
+		if !ok {
+			return false
+		}
+
+		return !reflect.DeepEqual(leaseOld.Spec.HolderIdentity, leaseNew.Spec.HolderIdentity)
+	}
+
 	return predicate.Funcs{
 		CreateFunc: func(event event.CreateEvent) bool {
 			return isSnapshotLease(event.Object)
 		},
 		UpdateFunc: func(event event.UpdateEvent) bool {
-			return isSnapshotLease(event.ObjectNew)
+			return isSnapshotLease(event.ObjectNew) && holderIdentityChange(event.ObjectOld, event.ObjectNew)
 		},
 		GenericFunc: func(event event.GenericEvent) bool {
```

Comment:
Why do you need to return true for `GenericFunc`? You do not do that in `JobStatusChanged` predicate.
Also as per the definition of `GenericEvent` I do not see why a `true` should ever be returned.
Refer: https://pkg.go.dev/sigs.k8s.io/controller-runtime@v0.16.3/pkg/event#GenericEvent
Currently we do not define any `Source` which generates events of this type. Please adjust the unit tests as well.

Reply:
Yes.. We can return false here

Reply:
Done